docs/participation_key_lifecycle.md

```diff
@@ -0,0 +1,162 @@
+# Participation Key Lifecycle
+
+This document goes into technical details about the lifecycle of participation
+keys. Before getting into that let's briefly discuss the purpose of these keys.
+
+## Overview
+Participation keys are used to participate in the consensus protocol. Aside
+from the registration process, they are completely decoupled from the account
+secret key.
+
+Each set of participation keys includes voting keys for each and every round
+for their validity range, in addition to state proof secrets for each and every
+state proof round (a state proof is not created every round). Because there
+are separate keys for each action we sometimes refer to the voting keys as
+**OneTimeKeys**. After each key is used, the secret is permenantly deleted from
+the node. This is a security property that eliminates the possibility of votes
+being changed in the event a node becomes compromised at a later date. This
+helps ensure finality even if a serious security breech were to occur.
+
+Aside from the **Registration** event, keys are completely decoupled from the
+account secrets. This is another security property. A securely managed account
+need only be exposed when signing the key registration transaction. For this
+reason tools like **algokey** are available for fully offline transaction
+signing.
+
+Given a set of participation keys, and a signed key registration transaction,
+an algod node can be configured to participation in the consensus protocol.
+For each round a single-use key is used to verify the authenticity of messages.
+
+## Key Creation
+
+Keys can be created with **goal** or **algokey**. There are a number of
+shortcuts and helpers, especially in **goal**, which wont be covered here.
+
+Required parameters are as follows:
+* **First**: the first round the keys will be valid for.
+* **Last**: the last round the keys will be valid for.
+* **Parent**: The public address of the account that will register the keys.
+
+The parent is stored along with the keys for convenience and error checking.
+It is not part of the cryptography. It can technically be set later, but must
+be set before the keys can be installed or registered.
+
+There is one optional parameter:
+* **Dilution**: Configure the multi-level key storage.
+
+Internally, keys are stored as a top-level array of keys that are used to
+generate the actual voting keys. The **Dilution** parameter defines how many
```

Comment:
Some questions I have about keys and the dilution parameter:
* No matter what the dilution is set to, is there still one unique voting key per round?
* What's the relationship between the "top-level array of keys" and the voting keys? If these top-level keys are all that are stored in the DB, how are the individual voting keys generated?

Personally I would find a visual helpful here. Maybe a mermaid diagram like this could be useful?

```mermaid
flowchart TD
    root["Participation Keys"] --> p0["Participation Key 0"]
    root --> p1["Participation Key 1"]
    root --> pDot["..."]
    root --> pN["Participation Key N"]
    p0 --> v0["Voting Key 0"]
    p0 --> v1["Voting Key 1"]
    p0 --> vDot["..."]
    p0 --> vN["Voting Key N"]
```

Reply:
Its only two levels. The top level is an array of public/private keys. To make the keys you remove the first top-level key and use it to make `key-dilution` voting keys.

top-level and voting are logical constructs, the actual keys for each are identical

Reply:
Added some clarifying text and a slightly modified version of this diagram.